preferred_assets_api/internal/adapters/repositories/inmemory/favourite_lru_repository.go

```diff
@@ -0,0 +1,141 @@
+package inmemory
+
+import (
+	"sync"
+	"time"
+
+	"github.com/MichailidouNatalia/GWI-Engineering-Challenge/preferred_assets_api/internal/adapters/repositories/entities"
+	"github.com/MichailidouNatalia/GWI-Engineering-Challenge/preferred_assets_api/internal/ports"
+	lru "github.com/hashicorp/golang-lru/v2"
+)
+
+var _ ports.FavouriteRepository = (*LRUFavouriteRepositoryImpl)(nil)
+
+type LRUFavouriteRepositoryImpl struct {
+	userAssetsCache *lru.Cache[string, map[string]time.Time]
+	existsCache     *lru.Cache[string, bool]
```

Comment:
I notice you have two caches here: userAssetsCache and existsCache. Can you walk me through your reasoning for having both? What problem does the existsCache solve?